Added password hashing to backend

jnorrick opened the pull request:
## IMPORTANT 
drop or truncate your customer table if you have customers already saved, the former plaintext passwords will no longer work

<br>

#### Added:
*BCrypt Password encoding and hashing functionality in LoginController and ProjectSecurityConfig
*PwHash setter to update the password from plain text to hashed in the process

#### Removed:
*dependencies not in use
*encoding from Customer model

<br>

#### How To Test:
Using Postman:
1. Open an new Postman request tab
2. Navigate to POST request for localhost:8080/register
3. place JSON object into body by selecting body radio button and add the below format for the JSON object (Populate the object with whatever strings you want for email, firstName, pwHash (min of 6 characters);
{
  "bookLibrary": null,
  "email": "professor@test.com",
  "firstName": "professor",
  "pwHash": "123456",
  "role": "user",
  "wishlist": null,
  "zipCode": 23456
}
5. this should add a new customer to the database when you view the customer, the password should be stored as the hashed values
6. Try adding a few different customers into the database so you can have multiple passwords to test
7. Feel free to test incorrect email formats, or leaving firstName, pwHash, or email blank (you should get errors as those are required fields)
8. Once you have a customer in the table, try to navigate to GET request for localhost:8080/testing. This route is still password protected. Once you get a 401, add your customer's email and password to the authorization tab using Basic Auth
9. Try the same GET request for localhost:8080/testing again. You should be able to now view the message "This should only be visible if Spring Security username and password works!"
10. feel free to try to add a book object as well. I didn't touch that, so it should still be working fine 😅

jnorrick committed:
Added BCrypt password encoding, removed unused dependencies from files, added additional validation on password field (now requires a min size of 6, max of max value)

dalpiad reviewed (changes requested):
Request validation with the Zip Code (restrict to 5 digits).  Recommend preventing the same email address from being used for multiple accounts.

jnorrick committed:
Added @Column Unique validation so that multiple users with the same email cannot be added to the database

jnorrick committed:
Adding annotation to zipcode with regex for 5 digits only

jnorrick commented:
> Request validation with the Zip Code (restrict to 5 digits). Recommend preventing the same email address from being used for multiple accounts.

* Added validation to prevent duplicate email 
* Added validation so that zip code must be 5 digits (May need to drop and re-create table in order for it to function!) Also, if needing to do any calculations with zip code or needing it to be an integer, you just need to cast it as such. The regex on it is looking for a sequence of 5 digits.

dalpiad reviewed (approved):
Looks great!  The zip code and email validation are working.
